FreeRTOS/Demo/WIN32-MingW/main.c

```diff
@@ -134,204 +159,221 @@ declared here, as a global, so it can be checked by a test that is implemented
+/* Thread handle for the keyboard input Windows thread. */
+static HANDLE xWindowsKeyboardInputThreadHandle = NULL;
+
+/* This stores the last key pressed that has not been handled.
+ * Keyboard input is retrieved by the prvWindowsKeyboardInputThread
+ * Windows thread and stored here. This is then read by the idle
+ * task and handled appropriately. */
+static int xKeyPressed = mainNO_KEY_PRESS_VALUE;
 
 /*-----------------------------------------------------------*/
 
 int main( void )
 {
-	/* This demo uses heap_5.c, so start by defining some heap regions.  heap_5
-	is only used for test and example reasons.  Heap_4 is more appropriate.  See
-	http://www.freertos.org/a00111.html for an explanation. */
-	prvInitialiseHeap();
-
-	/* Do not include trace code when performing a code coverage analysis. */
-	#if( projCOVERAGE_TEST != 1 )
-	{
-		/* Initialise the trace recorder.  Use of the trace recorder is optional.
-		See http://www.FreeRTOS.org/trace for more information. */
-		vTraceEnable( TRC_START );
-
-		/* Start the trace recording - the recording is written to a file if
-		configASSERT() is called. */
-		printf( "\r\nTrace started.\r\nThe trace will be dumped to disk if a call to configASSERT() fails.\r\n" );
-		printf( "Uncomment the call to kbhit() in this file to also dump trace with a key press.\r\n" );
-		uiTraceStart();
-	}
-	#endif
-
-	/* The mainCREATE_SIMPLE_BLINKY_DEMO_ONLY setting is described at the top
-	of this file. */
-	#if ( mainCREATE_SIMPLE_BLINKY_DEMO_ONLY == 1 )
-	{
-		main_blinky();
-	}
-	#else
-	{
-		main_full();
-	}
-	#endif
-
-	return 0;
+    /* Set interrupt handler for keyboard input. */
+    vPortSetInterruptHandler( mainINTERRUPT_NUMBER_KEYBOARD, prvKeyboardInterruptHandler );
+
+    /* Start keyboard input handling thread. */
+    xWindowsKeyboardInputThreadHandle = CreateThread(
+        NULL,                          /* Pointer to thread security attributes. */
+        0,                             /* Initial thread stack size, in bytes. */
+        prvWindowsKeyboardInputThread, /* Pointer to thread function. */
+        NULL,                          /* Argument for new thread. */
+        0,                             /* Creation flags. */
+        NULL);
+
+    fflush( stdout );
+
+    /* Use the cores that are not used by the FreeRTOS tasks. */
+    SetThreadAffinityMask( xWindowsKeyboardInputThreadHandle, ~0x01u );
+
+    /* This demo uses heap_5.c, so start by defining some heap regions.  heap_5
+    is only used for test and example reasons.  Heap_4 is more appropriate.  See
+    http://www.freertos.org/a00111.html for an explanation. */
+    prvInitialiseHeap();
+
+    /* Do not include trace code when performing a code coverage analysis. */
+    #if( projCOVERAGE_TEST != 1 )
+    {
+        /* Initialise the trace recorder.  Use of the trace recorder is optional.
+        See http://www.FreeRTOS.org/trace for more information. */
+        configASSERT( xTraceInitialize() == TRC_SUCCESS );

```

Comment:
Is there a reason for the swap to calling xTraceInitialize()  here instead of  the vTraceEnable( TRC_START ) call that we've been using?

Reply:
vTraceEnable is deprecated by the lib. If you look at where vTraceEnbale is defined it says its a deprecated call. so I've updated to the new calls.

Reply:
Swapping to the new "xTraceEnable" would make more sense in that case then right?
I'm mainly asking because [xTraceEnable()](https://github.com/FreeRTOS/FreeRTOS/blob/main/FreeRTOS-Plus/Source/FreeRTOS-Plus-Trace/trcSnapshotRecorder.c#L327) does more than just the [initialize call here](https://github.com/FreeRTOS/FreeRTOS/blob/main/FreeRTOS-Plus/Source/FreeRTOS-Plus-Trace/trcSnapshotRecorder.c#L1669)
If the trace.dump file it is making is working then I suppose that's what matters

Reply:
I call both. Technically xTraceEnable does both so I should only call it, but figured I'd stick with this as it feels like the more proper way to do it. Calling just xTraceEnable it could fail for a couple of reasons. If I call xTraceInitialize then xTraceEnable, I know that I failed to initialize rather than failed to start.